src/core/thread/mle_router.cpp

```diff
@@ -3198,6 +3202,11 @@ Neighbor *MleRouter::GetNeighbor(const Mac::ExtAddress &aAddress)
 
         for (int i = 0; i <= kMaxRouterId; i++)
         {
+            if (i == mRouterId)
+            {
+                continue;
+            }
```

Comment:
Oh, not related to problem or bug, just notice that we can skip the Router's Id when searching one of its neighbor with input address parameter. Although, input address parameter can be used to match what we want to find, that would help avoid extra `memcmp` by checking the Router's Id first, should be more efficient. Thought? Thanks.